src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoveryUtility.cs

```diff
@@ -10,11 +11,46 @@ public static class PluginDiscoveryUtility
     {
         public static Lazy<string> InternalPluginDiscoveryRoot { get; set; }
 
+        private static string NuGetPluginsDirectory = "NuGetPlugins";
+
         public static string GetInternalPlugins()
         {
-            var rootDirectory = InternalPluginDiscoveryRoot?.Value ?? System.Reflection.Assembly.GetEntryAssembly()?.Location;
+            return InternalPluginDiscoveryRoot?.Value ??
+                GetNuGetPluginsDirectoryRelativeToNuGetAssembly(typeof(PluginDiscoveryUtility).GetTypeInfo().Assembly.Location); // NuGet.*.dll
+        }
 
-            return rootDirectory ?? Path.GetDirectoryName(rootDirectory);
+#if IS_DESKTOP
+        /// <summary>
+        /// Given Visual Studio 2017 MSBuild.exe path, return the NuGet plugins directory which is in CommonExtensions/NuGetPluginsDirectory
+        /// </summary>
+        /// <param name="msbuildExePath">The MsBuildExe path. Needs to be a valid path. file:// not supported.</param>
+        /// <returns>The NuGet plugins directory, null if <paramref name="msbuildExePath"/> is null</returns>
+        /// <remarks>The MSBuild.exe is in MSBuild\15.0\Bin\MsBuild.exe, the NuGetPlugins directory is in Common7\IDE\CommonExtensions\Microsoft\NuGetPlugins</remarks>
+        public static string GetInternalPluginRelativeToMSBuildExe(string msbuildExePath)
+        {
+            var parentDirectory = "..";
+            return !string.IsNullOrEmpty(msbuildExePath) ?
+                PathUtility.GetAbsolutePath(
+                    Path.GetDirectoryName(msbuildExePath),
+                    Path.Combine(parentDirectory, parentDirectory, parentDirectory, "Common7", "CommonExtensions", "Microsoft", "NuGet", NuGetPluginsDirectory)
```

Comment:
are you missing "IDE" here?

`C:\Program Files (x86)\Microsoft Visual Studio\2017DogFood\Enterprise\Common7\IDE\CommonExtensions\Microsoft\NuGet`